ci/ubuntu-24.04/Dockerfile

```diff
@@ -36,8 +36,35 @@ RUN apt-get update && apt-get -y install \
     unzip \
     wget \
     zlib1g-dev \
+    # for custom ccache build \
+    libxxhash-dev \
+    libzstd-dev \
   && apt autoclean \
   && rm -rf /var/lib/apt/lists/*
 
 RUN pip3 install --break-system-packages junit2html
 RUN gem install coveralls-lcov
+
+# Download and build a newer ccache version that recognizes -fprofile-update=atomic
+# which is used when building with --coverage.
```

Comment:
Are we compiling our own version since we have to run this on platforms other than x86_64 also? I am asking since there is are [prebuilt binaries](https://ccache.dev/download.html) for that platform. If that is the reason let's at least capture it in a comment.

Reply:
Uhm, no reason other than I compiled from source to fix it locally for me and doing the same in the Dockerfile.

I've re-pushed the branch. Now uses the pre-built ccache version. wdyt?